### PR TITLE
Fixes json decode two times

### PR DESCRIPTION
**Pull request by joecohens**
Fixes https://github.com/dinkbit/notifyme/issues/4

//cc @GrahamCampbell

### CODE REVIEW

**Comment by joecohens** [comment]
Kicking StyleCI

**Comment by GrahamCampbell** [comment]
Errr. If the original job is still pending, styleci will just queue the next behind it. I'll check the queue processing now...

**Comment by GrahamCampbell** [comment]
You seem to have crashed it...

**Comment by GrahamCampbell** [comment]
https://i.starbs.net/AZWr

**Comment by joecohens** [comment]
Oh ok I thought it wasn't being processed.

**Comment by GrahamCampbell** [comment]
I'm killing the queue worker...

**Comment by GrahamCampbell** [comment]
The log is millions of lines long...

**Comment by joecohens** [comment]
Sorry :(

**Comment by GrahamCampbell** [comment]
lol, it wasn't the open close that caused it btw :)

**Comment by joecohens** [comment]
Lol what was?

**Comment by GrahamCampbell** [comment]
It seems to be the fact you've used an upper case I in the branch name.

**Comment by GrahamCampbell** [comment]
One sec while I investigate...

**Comment by joecohens** [comment]
Nice bug :)

**Comment by GrahamCampbell** [comment]
lol, I'll try running `git remote prune origin` in the cached git repo folder...

**Comment by GrahamCampbell** [comment]
errr, https://i.starbs.net/1ry2, i can't rememer which one is this repo, lol

**Comment by joecohens** [comment]
LOL

**Comment by GrahamCampbell** [comment]
It's dd22d4c7a395673687c289eeea3f34882e498f80. :)

**Comment by joecohens** [comment]
There you go :)

**Comment by GrahamCampbell** [comment]
Well that fixed it. Very odd. :/

**Comment by GrahamCampbell** [comment]
https://i.starbs.net/eoz6

**Comment by GrahamCampbell** [comment]
I think I need to get styleci to just delete the cloned repo and re-clone it if we get a git error. That would probably resolve such issues.

**Comment by joecohens** [comment]
That's weird so it wasn't the upper case?

**Comment by joecohens** [comment]
Yeah that would be cool, like a retry.

File: src/Dinkbit/Notifyme/Gateways/Twilio.php [hunk 1]
```diff
@@ -140,11 +140,11 @@ protected function parseResponse($body)
      */
     protected function responseError($rawResponse)
     {
-        if (! $this->isJson($rawResponse->getBody())) {
-            return $this->jsonError($rawResponse);
+        if ($json = $this->isJson($rawResponse->getBody())) {
```
Comment (GrahamCampbell) [inline src/Dinkbit/Notifyme/Gateways/Twilio.php]: Asignment inside if statements is generally regarded as bad. Symfony doesn't allow it for example.

**Comment by GrahamCampbell** [comment]
> That's weird so it wasn't the upper case?

No, I don't think it was. I have no idea what caused this.

File: src/Dinkbit/Notifyme/Gateways/Twilio.php (hunk 1 above)
Comment (joecohens) [inline src/Dinkbit/Notifyme/Gateways/Twilio.php]: I didn't know :)

``` php
$json = $this->isJson($rawResponse->getBody())

return $json ?: $this->jsonError($rawResponse);
```

Better?

Comment (GrahamCampbell) [inline src/Dinkbit/Notifyme/Gateways/Twilio.php]: If you do that, you could actually just do a one liner.

Comment (joecohens) [inline src/Dinkbit/Notifyme/Gateways/Twilio.php]: Done `return $this->isJson($rawResponse->getBody()) ?: $this->jsonError($rawResponse);`

File: src/Dinkbit/Notifyme/Gateways/HipChat.php [hunk 2]
```diff
@@ -193,13 +189,17 @@ public function jsonError($rawResponse)
      *
      * @param string $string
      *
-     * @return bool
+     * @return bool|array
      */
     protected function isJson($string)
```
Comment (GrahamCampbell) [inline src/Dinkbit/Notifyme/Gateways/HipChat.php]: isJson isn't really the correct method name now

File: src/Dinkbit/Notifyme/Gateways/Slack.php [hunk 3]
```diff
@@ -167,13 +163,17 @@ public function jsonError($rawResponse)
      *
      * @param string $string
      *
-     * @return bool
+     * @return bool|array
```
Comment (GrahamCampbell) [inline src/Dinkbit/Notifyme/Gateways/Slack.php]: array|bool would be better :)

File: src/Dinkbit/Notifyme/Gateways/HipChat.php (hunk 2 above)
Comment (joecohens) [inline src/Dinkbit/Notifyme/Gateways/HipChat.php]: That's right it's not returning a bool only anymore.. `validateJson()` or `isValidJson`?

Comment (GrahamCampbell) [inline src/Dinkbit/Notifyme/Gateways/HipChat.php]: getJson() would be better surely?

Comment (GrahamCampbell) [inline src/Dinkbit/Notifyme/Gateways/HipChat.php]: and doesn't json_decode return null on error anyway, so this entire method is a bit over kill

Comment (joecohens) [inline src/Dinkbit/Notifyme/Gateways/HipChat.php]: Lol yeah!

Comment (joecohens) [inline src/Dinkbit/Notifyme/Gateways/HipChat.php]: Removed completely :)

**Comment by joecohens** [comment]
Done. Merging this thank you very much sir!  :)

**Comment by GrahamCampbell** [comment]
:)

**Comment by GrahamCampbell** [comment]
I'm working on creating a decorator for the repository class in styleci fixer that will re-entry stuff and delete the repo if it's damaged ect.

**Comment by joecohens** [comment]
Awesome!!!!

**Comment by GrahamCampbell** [comment]
I think we might even need to move this stuff to a new package. It's getting quite large now...

**Comment by GrahamCampbell** [comment]
Namespaces are properly massive now, lol: https://i.starbs.net/AW67.

**Comment by joecohens** [comment]
LOL really massive and hard to write :)

**Comment by GrahamCampbell** [comment]
Hmm. I'm going to drop "Failed" from the exception names. I'm also going to move the "StyleCI\Fixer\Git" namespace to "StyleCI\Git" at a later point too. :)

**Comment by joecohens** [comment]
:+1: Yeah I like it better.

**Pull request merged**